mne/viz/raw.py

```diff
@@ -163,7 +164,56 @@ def _mouse_click(event, params):
-        _pick_bad_channels(event, params)
+        # vertical lines added
+        if event.key == 'shift':
+            scrl = params['ax_hscroll']
+            if (params['fillarea'] in params['ax'].collections):
+                params['ax'].collections.remove(params['fillarea'])
+                scrl.collections.remove(params['hscroll_fillarea'])
+            x = np.array([event.xdata] * 2)
+            y = np.array(params['ax'].get_ylim())
+            if event.button == 1:
+                xdata = params['ax_leftline'].get_xdata()
+                if np.abs(xdata[0] - x[0]) < 0.2:
+                    params['ax_leftline'].set_data([0, 0], y)
+                    params['leftline_t'].set_text('')
+                    params['ax_hscroll_leftline'].set_data([0, 0], [0, 1])
+                    event.canvas.draw()
+                    return
+                params['ax_leftline'].set_data(x, y)
+                params['leftline_t'].set_text('%0.3f' % x[0])
+                params['ax_hscroll_leftline'].set_data(x, np.array([0., 1.]))
+                xdata = params['ax_rightline'].get_xdata()
+                params['fillarea'] = params['ax'].fill_betweenx(y, x, xdata,
+                                                                alpha=0.2,
+                                                                facecolor='y')
+                params['hscroll_fillarea'] = scrl.fill_betweenx([0, 1], x,
+                                                                xdata,
+                                                                alpha=0.2,
+                                                                facecolor='y')
+            elif event.button == 3:
+                xdata = params['ax_rightline'].get_xdata()
+                if np.abs(xdata[0] - x[0]) < 0.2:
+                    x = params['raw'].index_as_time(params['raw'].last_samp)[0]
+                    params['ax_rightline'].set_data([x, x], y)
+                    params['rightline_t'].set_text('')
+                    params['ax_hscroll_rightline'].set_data([x, x], [0, 1])
+                    event.canvas.draw()
+                    return
+                params['ax_rightline'].set_data(x, y)
+                params['rightline_t'].set_text('%0.3f' % x[0])
+                params['ax_hscroll_rightline'].set_data(x, [0, 1])
+                xdata = params['ax_leftline'].get_xdata()
+                params['fillarea'] = params['ax'].fill_betweenx(y, xdata, x,
+                                                                alpha=0.2,
+                                                                facecolor='y')
+                params['hscroll_fillarea'] = scrl.fill_betweenx([0, 1], x,
+                                                                xdata,
+                                                                alpha=0.2,
+                                                                facecolor='y')
+            event.canvas.draw()
```

Comment:
This should probably all be put into a private function like `_pick_bad_channels` is to keep this logical separation clean